roles/docker/tasks/main.yml

```diff
@@ -7,14 +7,19 @@
   register: docker_version_result
   changed_when: false
 
+- name: Remove deferred deletion for downgrades from 1.9
+  command: >
+    sed -i 's/--storage-opt dm.use_deferred_deletion=true//' /etc/sysconfig/docker-storage
+  when: not docker_version_result | skipped and docker_version_result.stdout | default('0.0', True) | version_compare('1.9', '>=') and docker_version | version_compare('1.9', '<')
```

Comment:
Maybe it's just github but it looks like you didn't remove the old version of this from below? We also need the check for the file existence moved above this, i'd drop its when clause all together.